src/hooks/UseQuery.re

```diff
@@ -56,25 +64,50 @@ let useQuery =
   let variables = request##variables;
   let parse = request##parse;
   let rp =
-    UseSemanticGuarantee.useSemanticGuarantee(
-      () => Belt.Option.map(requestPolicy, Types.requestPolicyToJs),
-      requestPolicy,
+    React.useMemo1(
+      () => requestPolicy->Belt.Option.map(Types.requestPolicyToJs),
+      [|requestPolicy|],
     );
-  let context = {
-    Types.partialOperationContext(
-      ~additionalTypenames?,
-      ~fetchOptions?,
-      ~fetch?,
-      ~url?,
-      ~pollInterval?,
-      ~meta?,
-      ~suspense?,
-      ~preferGetMethod?,
-      (),
+
+  let client = UseClient.useClient();
+
+  let context =
```

Comment:
So this is a big bummer. Without this memoization, we create a new `context` object on every render, which causes `useQuery` to run, which re-renders, which generates a new `context` object, and boom, we're off to infinite re-renders. So we have to memoize the `context` object. Unfortunately `reason-react` only provides `useMemo` arity up to 7 so I had to bring in `useMemo10` 😂